src/dfx/src/lib/config.rs

```diff
@@ -6,10 +6,19 @@ use std::path::PathBuf;
 
 #[context("Failed to get path to dfx config dir.")]
 pub fn get_config_dfx_dir_path() -> DfxResult<PathBuf> {
-    let config_root = std::env::var("DFX_CONFIG_ROOT").ok();
-    let home = std::env::var("HOME").context("Failed to resolve 'HOME' env var.")?;
-    let root = config_root.unwrap_or(home);
-    let p = PathBuf::from(root).join(".config").join("dfx");
+    let config_root = std::env::var_os("DFX_CONFIG_ROOT");
+    // dirs-next is not used for *nix to preserve existing paths
+    #[cfg(not(windows))]
+    let p = {
+        let home = std::env::var_os("HOME").context("Failed to resolve 'HOME' env var.")?;
+        let root = config_root.unwrap_or(home);
+        PathBuf::from(root).join(".config").join("dfx")
+    };
+    #[cfg(windows)]
+    let p = config_root.map_or_else(
+        || dirs_next::config_dir().unwrap().join("dfx"),
+        PathBuf::from,
+    );
```

Comment:
Suggest using `ProjectDirs::from("org", "dfinity", "dfx")` from directories_next here, so that on Windows these files will sit in the same directory as networks.json